iOS/Collection View Management /DiffableCollectionViewController.swift

```diff
@@ -0,0 +1,129 @@
+//
+//  DiffableCollectionViewController.swift
+//  Jibber
+//
+//  Created by Benji Dodgson on 9/25/21.
+//  Copyright © 2021 Benjamin Dodgson. All rights reserved.
+//
+
+import Foundation
+
+class newDataSource<SectionType: Hashable, ItemType: Hashable>: CollectionViewDataSource<SectionType, ItemType> {
+
+}
+
+class DiffableCollectionViewController<SectionType: Hashable, ItemType: Hashable, DataSource: newDataSource<SectionType, ItemType>>: ViewController, UICollectionViewDelegate {
+    
+    lazy var dataSource = DataSource.init(collectionView: self.collectionView)
+
+    @Published var selectedItems: [ItemType] = []
```

Comment:
This is just really nice to have. Instead of maintaining an array, we just use the internal collectionview one and apply it to this published property.